### DISCUSSION TITLE
Get the rendered html from a widget's children.

### DISCUSSION THREAD

**Discussion by GameDungeon** [TiddlyWiki/TiddlyWiki5#7578] [Q&A]
I want to make a widget that gets and then edits the rendered HTML output of all of its children. I notice the render functions don't return the HTML, so I'm not sure how to get the child element's HTML. Any tips on how to approach it?

**Comment by pmario** [comment]
Do you want wikitext or javascript info?

**Reply by GameDungeon** [reply]
Info on how to do it in wiki or JS? I want to do it in JS

**Reply by buggyj** [reply]
in the render function usually there is 

```
this.makeChildWidgets();
this.renderChildren(parent,nextSibling);
```

after the call to `renderChildren` you can add 

`childHtml = parent.innerHTML;`
